packages/tools/client-logger/app-insights-logger/src/FluidAppInsightsLogger.ts

```diff
@@ -0,0 +1,53 @@
+import { ApplicationInsights, Snippet } from "@microsoft/applicationinsights-web";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+/**
+ * Config object for FluidAppInsightsLogger.
+ */
+export interface FluidAppInsightsLoggerConfig {
+	/**
+	 * Accepts either an existing preconfigured ApplicationInsights instance to use
+	 * for logging or a config object used to instantiate an instance of ApplicationInsights. If providing a
+	 * preconfigured instance of ApplicationInsights, .loadAppInsights() must be called on the instance first.
+	 */
+	appInsights: ApplicationInsights | Snippet;
+}
+
+export class FluidAppInsightsLogger implements ITelemetryBufferedLogger {
+	protected readonly baseLoggingClient: ApplicationInsights;
+
+	/**
+	 * Creates an instance of FluidAppInsightsLogger using either a provided instance of ApplicationInsights
+	 * logging client or by creating a new instance with a provided ApplicationInsights configuration object.
+	 */
+	public constructor(config: FluidAppInsightsLoggerConfig) {
+		if (config.appInsights instanceof ApplicationInsights) {
```

Comment:
instanceof should never be used cross package or on exported types. it can only be used safely for types within a package. this is because there can be Mutiple instances of a package used by different layers of the stack, and instanceof will not recognize them as the same.

Reply:
i would probably drop the union, which isn't really safe, and jsut provide two free or static methods that explicitly take the types, and not try to infer anything

Reply:
updated this class to simply accept the app insights client for its constructor in commit 70aa9fd